### ISSUE TITLE
Add "Show members" option to circle options

### ISSUE THREAD

**Issue by gibelium** [nextcloud/circles#1431]
Within the circle settings you can choose the option "Visible to everyone" which will display a circle in the list of available circles for users, that are not members of this circle. When a non-member selects this circle it will display "The list of members is only visible to members of this circle" at the members area.

I would suggest to add a second config option "Show members" in the visibility section which would show the members of a circle to users that are not yet members when the "Visible to everyone" option is active.

**Comment by flopdidopp** [comment]
This would be very handy. We do need this feature. An addition would very much appreciated.

**Comment by andynameistaken** [comment]
Yes it would be really helpful especially for work that relies on collaboration between different departament  / team. Currently there is no elegant way to have quick view who belongs to what, the only way to see members of all teams is where everyone belongs to every group which defeats its purpose.

**Comment by andynameistaken** [comment]
I really want to recommend Nextcloud for collaboration over discord + google suite for different entities, but it is really hard to do it, because everyone is used to those two things which people consider basic now: being able to take a quick look of groups that user belongs to and pinning messages in chat. People are turning away, because they either think Nextcloud is half-baked or stiff. Implementing those features would greatly improve social image of the product.

**Comment by gibelium** [comment]
I still think the feature is super important and I'm surprised that few people are interested in it. For good collaboration between teams and departments, it is essential to see who is part of another team that you don't belong to and who you can contact if you need something from this team.

I offer to implement the feature and would like to discuss with those responsible how this can be best implemented.

**Comment by gibelium** [comment]
It seems that this feature has to be implemented within the contacts app and there already exists a ticket for this in the related repository.

https://github.com/nextcloud/contacts/issues/4097

**Comment by operthuis** [comment]
Current behavior:
Within the circle settings, you can enable the option "Visible to everyone", which displays the circle in the list of available circles to users who are not members.
However, when a non-member opens the circle, the members list remains hidden, and the message “The list of members is only visible to members of this circle” is displayed.

Suggestion:
I propose adding an additional visibility option — for example, “Show members to non-members” — within the circle’s settings.
When “Visible to everyone” is enabled, this second option would allow administrators to choose whether the list of members should also be visible to non-members.

Use case / benefit:
This would make it easier for users to identify team members before deciding to join a circle. It adds flexibility, particularly for open or semi-open collaboration environments where transparency is important.
